test/e2e/federated-ingress.go

```diff
@@ -104,7 +105,11 @@ var _ = framework.KubeDescribe("Federated ingresses [Feature:Federation]", func(
 			defer func() { // Cleanup
 				By(fmt.Sprintf("Deleting ingress %q in namespace %q", ingress.Name, ns))
 				err := f.FederationClientset_1_5.Ingresses(ns).Delete(ingress.Name, &v1.DeleteOptions{})
-				framework.ExpectNoError(err, "Error deleting ingress %q in namespace %q", ingress.Name, ns)
+				framework.ExpectNoError(err, "Error deleting ingress %q/%q in federation", ns, ingress.Name)
+				for clusterName, cluster := range clusters {
+					err := cluster.Ingresses(ns).Delete(ingress.Name, &v1.DeleteOptions{})
```

Comment:
Also, I think you need to do something akin to `cleanupServiceShardsAndProviderResources()` (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/federated-ingress.go#L139) to ensure the underlying cloud resources are cleaned up. 

For an Ingress specific example, see the Ingress e2e test - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/ingress_utils.go#L311

But let's get this PR in first.